app/src/main/java/com/mealsmadeeasy/ui/home/discover/SearchActivity.kt

```diff
@@ -78,6 +84,11 @@ class SearchActivity : BaseActivity() {
                 return true
             }
         })
+
+        if (querySubject.value != null && querySubject.value != "") {
+            searchView.setQuery(querySubject.value as String, false)
+        }
+
```

Comment:
I'd write this as `querySubject.value?.takeIf { it.isNotEmpty() }?.let{ searchView.setQuery(it, false) }`. Thoughts?